crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -242,19 +250,27 @@ impl DnsHandle for GenericConnection {
 #[derive(Clone)]
 pub struct GenericConnector<P: RuntimeProvider> {
     runtime_provider: P,
+    #[cfg(feature = "dns-over-rustls")]
+    client_config: Arc<Lazy<Result<Arc<rustls::ClientConfig>, ProtoError>>>,
 }
 
 impl<P: RuntimeProvider> GenericConnector<P> {
     /// Create a new instance.
     pub fn new(runtime_provider: P) -> Self {
-        Self { runtime_provider }
+        Self {
+            runtime_provider,
+            #[cfg(feature = "dns-over-rustls")]
+            client_config: Arc::new(Lazy::new(crate::tls::client_config)),
+        }
     }
 }
 
 impl<P: RuntimeProvider + Default> Default for GenericConnector<P> {
     fn default() -> Self {
         Self {
             runtime_provider: P::default(),
+            #[cfg(feature = "dns-over-rustls")]
+            client_config: Arc::new(Lazy::new(crate::tls::client_config)),
```

Comment:
I might have missed it, but I'm having trouble following the advantage of this lazy in this context, i.e. it seems like this will be a new reference, so even the Arc is unnecessary? Am I misunderstanding this?

Reply:
The `Arc` here is required to clone the `Lazy`, which is needed if we want to keep `GenericConnector` cloneable.
`Lazy` is used here to lazily initialize the `ClientConfig`, because this config isn't used if the user supplies it's own.